cart/templates/cart.html

```diff
@@ -100,12 +127,29 @@
     return false;
 }
 
+function add_discount(event) {
+    var discount_code = encodeURIComponent(document.getElementById("discount_code").value);
+    var csrfmiddlewaretoken = encodeURIComponent(document.querySelector("input[name='csrfmiddlewaretoken']").value);
+    var pathArray = window.location.pathname.split('/');
```

Comment:
nit: here and in the basket clear code, we can just put the slug in here from the template rather than parsing it out